server/mdm/nanomdm/service/nanomdm/service.go

```diff
@@ -232,7 +232,19 @@ func (s *Service) CommandAndReportResults(r *mdm.Request, results *mdm.CommandRe
 	logger.Info(logs...)
 	err := s.store.StoreCommandReport(r, results)
 	if err != nil {
-		return nil, fmt.Errorf("storing command report: %w", err)
+		// allow not found commands, this is an edge case only
+		// valid for migrations, other response codes confuse the
+		// mdmclient, and this gives us the opportunity to answer
+		// with more commands
+		if !service.IsNotFound(err) {
+			return nil, fmt.Errorf("storing command report: %w", err)
+		}
+
+		logger.Debug(
```

Comment:
Should we also log the reported status and any error fields here? Would it make sense to log this at the `Error` level instead of `Debug`?

Reply:
so, debug makes more sense to me, but I can change it and add the other fields if you think it'd be helpful

Reply:
I do think that having the extra fields would be helpful info that we would lose if we aren't logging it.

Debug makes sense too. My only worry is that we lose the report altogether if instances don't have debug logging enabled. I believe we have debug logging enabled for our cloud-hosted customers, so it's probably not a major concern.

Reply:
so, there's no error on the logger, I added it as `Info` and added both fields